test/e2e/gateway/008_tlsroute_test.go

```diff
@@ -27,16 +27,16 @@ import (
 	gatewayv1alpha1 "sigs.k8s.io/gateway-api/apis/v1alpha1"
 )
 
-func testTLSRoute(fx *e2e.Framework) {
+func testTLSRoute(fx *e2e.Framework, namespace string) {
```

Comment:
we could make this `testTLSRoutePassthrough` and have another that is `testTLSRouteTerminate` for the other case maybe? run a plain HTTP server behind contour, connect on HTTPS and our request should go to that service

Reply:
Yeah we can do that, was just trying to minimize the number of duplicate resources, but I can pull out the TLS route and reference from both since these tests are really testing the gateway configuration, not the TLSRoute configuration, but that will need to change as we add more.

Reply:
I ended up taking out the terminate test. We don't support it right now and also don't set status. 

This is happening because when the TLSRoute is selected by an invalid Gateway, the route is not processed (since the Gateway is bad), but need to set an "Admitted: false" for the selected routes probably, but not yet implemented.

Reply:
> I ended up taking out the terminate test. We don't support it right now and also don't set status.

Could you clarify this? Looks like we're implementing some of it here: https://github.com/projectcontour/contour/blob/d5d9e55050d85c0793ff6daf8e3726bd7c3ba8bd/internal/dag/gatewayapi_processor.go#L101

are we just doing validation for now and still doing passthrough for all TLSRoutes?

Reply:
i see we're still not attaching the TLS secret to the dag secure vhost etc. https://github.com/projectcontour/contour/blob/d5d9e55050d85c0793ff6daf8e3726bd7c3ba8bd/internal/dag/gatewayapi_processor.go#L447

Reply:
should we explicitly say Terminate mode is not allowed for now and enable it in the PR that provides support for that?

Reply:
Yeah this doesn't add Terminate mode, just some more plumbing to make it useful. It was my understanding originally chatting with @youngnick that TLSRoute **only** did passthrough, but that's not the case, so this is gradually adding more to enable Terminate mode in the next PR.